src/Microsoft.SqlTools.ServiceLayer/QueryExecution/SpecialAction.cs

```diff
@@ -0,0 +1,132 @@
+        
+        #region Private Class variables 
+        bool none;
+        bool expectActualYukonXmlShowPlan;
+        bool expectEstimatedYukonXmlShowPlan;
+
+        #endregion
+
+        /// <summary>
+        /// a type of XML execution plan may be returned  
+        /// </summary>
+        public SpecialAction()
+        {
+            None = true;
+            ExpectActualYukonXmlShowPlan = false;
+            ExpectEstimatedYukonXmlShowPlan = false;
+        }
+
+        #region Public Functions
+        /// <summary>
+        /// No Special action performed 
+        /// </summary>
+        public bool None {
+            get { return none; }
+            set { 
+                none = value;
+                if (value)
+                {
+                    expectActualYukonXmlShowPlan = false;
+                    expectEstimatedYukonXmlShowPlan = false;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Actual XML execution plan may be returned 
+        /// </summary>
+        public bool ExpectActualYukonXmlShowPlan 
+        {
+            get { return expectActualYukonXmlShowPlan; }
+            set { this.registerSpecialAction(ref expectActualYukonXmlShowPlan, value); }
+        }
+
+        /// <summary>
+        /// Estimated XML execution plan may be returned  
+        /// </summary>
+        public bool ExpectEstimatedYukonXmlShowPlan 
+        {
+            get { return expectEstimatedYukonXmlShowPlan; }
+            set { this.registerSpecialAction(ref expectEstimatedYukonXmlShowPlan, value); }
+        }
+
+        /// <summary>
+        /// a type of XML execution plan may be returned  
+        /// </summary>
+        public bool ExpectYukonXMLShowPlan 
+        {
+            get { return ExpectEstimatedYukonXmlShowPlan || ExpectActualYukonXmlShowPlan; }
+            set 
+            { 
+                ExpectEstimatedYukonXmlShowPlan = value;
+                ExpectActualYukonXmlShowPlan = value;
+            }
+        }
+
+        /// <summary>
+        /// Aggregate this special action with another one  
+        /// </summary>
+        public void CombineSpecialAction(SpecialAction action)
+        {
+            if (!action.None)
+            {   
+                this.None = false;
+                
+                if (action.ExpectActualYukonXmlShowPlan) 
+                {
+                    this.ExpectActualYukonXmlShowPlan = true;
+                }
+
+                if (action.ExpectEstimatedYukonXmlShowPlan) 
+                {
+                    this.ExpectEstimatedYukonXmlShowPlan = true;
+                }
+            }
+        }
+        
+        #endregion
+
+        #region Private Helper Functions 
+        /// <summary>
+        /// Check to see if all properties are false, other than none 
+        /// </summary>
+        private bool areAllFalse()
+        {
+            if (!ExpectActualYukonXmlShowPlan && !ExpectEstimatedYukonXmlShowPlan)
```

Comment:
>!ExpectActualYukonXmlShowPlan && !ExpectEstimatedYukonXmlShowPlan [](start = 16, length = 65)

method could be rewritten as "return !ExpectActualYukonXmlShowPlan && !ExpectEstimatedYukonXmlShowPlan;"